client/src/app/core/ui-services/linenumbering.service.ts

```diff
@@ -135,14 +135,16 @@ export class LinenumberingService {
      * @returns {string}
      */
     public djb2hash(str: string): string {
-        let hash = 5381;
-        let char;
-        for (let i = 0; i < str.length; i++) {
-            char = str.charCodeAt(i);
-            // tslint:disable-next-line:no-bitwise
-            hash = (hash << 5) + hash + char;
+        if (str) {
```

Comment:
Now the empty string gets another hash as before. is this wanted? Also the same as above: Correct the return type and add null checks everywhere this function is called.